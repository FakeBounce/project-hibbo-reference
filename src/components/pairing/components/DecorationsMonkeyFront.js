import React, { PureComponent } from 'react';
import { Animated, Easing } from 'react-native';
import { bool } from 'prop-types';

import leftTree2Image from 'assets/pairing/monkey/leftTree2.png';
import bottomTreeImage from 'assets/pairing/monkey/bottomTree.png';

import * as Styles from '../styles';

class DecorationsMonkeyFront extends PureComponent {
  state = {
    leftTree2: new Animated.Value(0),
    bottomTree: new Animated.Value(0),
    leftX: new Animated.Value(0),
    bottomY: new Animated.Value(0),
  };

  componentDidMount() {
    const { leftTree2, bottomTree } = this.state;

    if (this.props.animated) {
      Animated.parallel([
        Animated.timing(bottomTree, {
          toValue: -125,
          delay: 50,
          easing: Easing.in(Easing.quad),
          useNativeDriver: true,
        }),
        Animated.timing(leftTree2, {
          toValue: 138,
          delay: 125,
          easing: Easing.in(Easing.quad),
          useNativeDriver: true,
        }),
      ]).start();
    } else {
      leftTree2.setValue(138);
      bottomTree.setValue(-125);
    }
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.reset !== nextProps.reset && nextProps.reset) {
      this.resetAnims();
    }
  }

  resetAnims() {
    const { leftX, bottomY } = this.state;

    Animated.parallel([
      Animated.timing(bottomY, {
        toValue: 400,
      }),
      Animated.timing(leftX, {
        toValue: -200,
      }),
    ]).start();
  }

  render() {
    const { bottomTree, leftTree2, leftX, bottomY } = this.state;

    return (
      <Styles.AZIndexFront>
        <Styles.AWrapperFront
          style={{
            transform: [
              {
                translateX: leftX,
              },
            ],
          }}
        >
          <Styles.ALeftTreeFront
            style={{
              transform: [
                {
                  translateX: leftTree2,
                },
              ],
            }}
            source={leftTree2Image}
          />
        </Styles.AWrapperFront>
        <Styles.AWrapperFront
          style={{
            transform: [
              {
                translateY: bottomY,
              },
            ],
          }}
        >
          <Styles.ABottomTree
            style={{
              transform: [
                {
                  translateY: bottomTree,
                },
              ],
            }}
            source={bottomTreeImage}
          />
        </Styles.AWrapperFront>
      </Styles.AZIndexFront>
    );
  }
}

DecorationsMonkeyFront.propTypes = {
  animated: bool.isRequired,
  reset: bool.isRequired,
};

export default DecorationsMonkeyFront;
