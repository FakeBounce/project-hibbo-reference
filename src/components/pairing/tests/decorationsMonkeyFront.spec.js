import React from 'react';
import renderer from 'react-test-renderer';

import DecorationsMonkeyFront from '../components/DecorationsMonkeyFront';

describe('DecorationsMonkeyFront', () => {
  const component = renderer.create(
    <DecorationsMonkeyFront
      animated
      reset={false}
    />,
  ).toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
