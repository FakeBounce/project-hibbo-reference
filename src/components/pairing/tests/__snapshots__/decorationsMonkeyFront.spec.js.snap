// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecorationsMonkeyFront should render the component 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "transform": Array [
        Object {
          "translateX": 0,
        },
      ],
      "zIndex": 2,
    }
  }
>
  <Image
    collapsable={undefined}
    source={1}
    style={
      Object {
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  />
</View>
`;
